### PR TITLE
feat(scExpression): added cell count to scExpression

### PR DESCRIPTION
**Pull request by atarashansky**
- #2696

### Reviewers
**Functional:** 
@hthomas-czi 
@signechambers1 
@seve 

image

### CODE REVIEW

**Review by seve — COMMENTED** [review]
TS looks good to me, could you add a python reviewer?

**Comment by atarashansky** [comment]
@MDunitz can you take a look at the backend changes? Turns out that what I had done previously was incorrect as the `joined` dataframe that was being iterated through was the output of `groupby(...).first()`, so it didn't actually capture the total number of cells.

File: backend/wmg/api/v1.py [hunk 1]
```diff
@@ -128,7 +130,7 @@ def build_expression_summary(query_result: DataFrame) -> dict:
     return structured_result
 
 
-def build_dot_plot_matrix(query_result: DataFrame, cell_counts: DataFrame) -> DataFrame:
+def build_dot_plot_matrix(query_result: DataFrame, cell_counts: DataFrame) -> Tuple[DataFrame, DataFrame]:
```
Comment (MDunitz) [inline backend/wmg/api/v1.py:133]: Can you update this function name and description to match the added functionality?
or even better refactor to  create a new function that calls `build_dot_plot_matrix` and a separate function calling fetching the cell counts

File: backend/wmg/api/v1.py [hunk 2]
```diff
@@ -142,9 +144,12 @@ def build_dot_plot_matrix(query_result: DataFrame, cell_counts: DataFrame) -> Da
     cell_counts_tissue_agg = cell_counts.groupby(["tissue_ontology_term_id"], as_index=True).sum()
     cell_counts_tissue_agg.rename(columns={"n_total_cells": "n_cells_tissue"}, inplace=True)
 
-    return expr_summary_agg.join(
-        cell_counts_cell_type_agg, on=["tissue_ontology_term_id", "cell_type_ontology_term_id"], how="left"
-    ).join(cell_counts_tissue_agg, on=["tissue_ontology_term_id"], how="left")
+    return (
+        expr_summary_agg.join(
```
Comment (MDunitz) [inline backend/wmg/api/v1.py:148]: can you please add a unit test specifically testing this logic

File: tests/unit/backend/wmg/api/test_v1.py [hunk 3]
```diff
@@ -444,27 +444,27 @@ def test__query_total_cell_count_per_cell_type(self, load_snapshot, ontology_ter
...
                         "cell_type_ontology_term_id": "cell_type_ontology_term_id_0",
-                        "total_count": expected_count,
+                        "total_count": expected_count * 64,
                         "depth": 0,
                     },
                     {
                         "cell_type": "cell_type_ontology_term_id_1_label",
                         "cell_type_ontology_term_id": "cell_type_ontology_term_id_1",
-                        "total_count": expected_count,
+                        "total_count": expected_count * 64,
```
Comment (MDunitz) [inline tests/unit/backend/wmg/api/test_v1.py:467]: Please add an explanation for the `* 64`

File: tests/unit/backend/wmg/api/test_v1.py [hunk 4]
```diff
@@ -186,6 +186,15 @@ def test__query_request_multi_primary_dims_only__returns_200_and_correct_respons
             )
             assert expected_cell_count_per_cell_type == 729
 
+            # there are 729 possible combinations per tissue-cell type given the above filtering criteria,
```
Comment (MDunitz) [inline tests/unit/backend/wmg/api/test_v1.py:189]: thank you!

**Review by MDunitz — APPROVED** [review]
lgtm

File: frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/style.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@ export const XAxisMask = styled.div`
   width: ${Y_AXIS_CHART_WIDTH_PX}px;
   position: sticky;
   left: 0;
+  opacity: 0;
```
Comment (tihuan) [inline frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/style.ts:20]: Ah sorry the opacity is needed here for the sticky header behavior when the user adds enough genes and scrolling starts to happen!

To debug, we can make overflow happen sooner with less added genes by increasing the screen size with `cmd` + `+`

mask opacity

Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/style.ts:20]: Currently the `Cell Count` header is not sticky while the values are. I'm assuming the product requirement is wanting both to be sticky when scrolling?

If so, I'm thinking maybe we can create a new DOM element to display the Cell Count column without relying on echarts, and make sure the new element still covers the gene labels when scrolling happens?

**Pull request merged**